server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -377,10 +399,10 @@ public void insert(InsertPlan insertPlan) throws QueryProcessException {
       latestFlushedTimeForEachDevice.putIfAbsent(batchInsertPlan.getDeviceId(), Long.MIN_VALUE);
 
       Integer[] results = new Integer[batchInsertPlan.getRowCount()];
-      List<Integer> sequenceIndexes = new ArrayList<>();
-      List<Integer> unsequenceIndexes = new ArrayList<>();
-
       long lastFlushTime = latestFlushedTimeForEachDevice.get(batchInsertPlan.getDeviceId());
+
+      HashMap<Long, List<Integer>> sequenceTimeRangeIndexes = new HashMap<>();
+      HashMap<Long, List<Integer>> unsequenceTimeRangeIndexes = new HashMap<>();
```

Comment:
Let me try to refactor this. After refactor, I will comment this block again. XD

Reply:
After rethinking this issue. I think this refactor should be a new pr because it's not related to storage group division. Before this pr, we also can refactor batch insert to get a better performance. So, I want to refactor this issue in next pr.

Reply:
Good, just keep it in mind.